### ISSUE TITLE
Skip long titled folder?

### ISSUE THREAD

**Issue by scriptzteam** [ncw/rclone#762]
```
Received error: failed to make directory: HTTP code 400: "400 Bad Request", reponse body: {"message":"1 validation error detected: Value 'BLABLAlooooNG.....ENCRYPTED' at 'name' failed to satisfy constraint: Member must have length less than or equal to 280"}
 - low level retry 1/10
```

is there a ways to skip that?

**Comment by ncw** [comment]
You could probably put a filter rule in.

if you did `--exclude="????...????*"` where you need about 175 `?` symbols then that will exclude any files which are too long after crypt.  You'd need another one for directories too, so I'd probably put them in an `--exclude-from` file with these two lines in

```
???????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????*
???????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????*/
```

Sorry that isn't very elegant but it is the best I can come up with at the moment!

Which remote storage system are you using?  The rules for name lengths vary for all of them.

**Comment by scriptzteam** [comment]
Its Amazon Cloud Drive with CRYPT to be exact :)

**Comment by camjac251** [comment]
I too am having this issue. Is there a way to have it not skip the files? I have the same config as scriptzteam with ACD and crypt setup, so crypt is causing the length to go past 280 right? My original path is only 205 characters long.

**Comment by ncw** [comment]
@camjac251 the right solution is #637 which will solve this problem properly.  Until then there isn't a nice solution :-(

**Comment by DurvalMenezes** [comment]
Hi, 

> if you did --exclude="????...????*" where you need about 175 ? symbols then that will exclude any files which are too long after crypt.

For the record, that didn't work for me: I still had many files that provoked the error above (using Amazon Cloud Drive with rclone encryption). 

I determined empiricaly that the maximum length in my situation is 160 characters; anything above that and I got errors. 

So, for me, the two exclude lines that worked were:

```
????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????*
????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????*/
```

Hope that helps. 

Cheers, 
-- 
   Durval.

**Comment by ncw** [comment]
That really is very ugly isn't it!

I'll make a new flag `--max-name-length` which you can use to filter this explicitly

**Comment by arionl** [comment]
Would it be possible to make this somewhat deterministic based on cloud provider configuration? A pretty common use case is using the crypt overlay on top of a normal provider (I'm doing that with AmazonDrive). So if AmazonDrive is limited to a certain length (as documented in the API) and layered crypt further limits this, could rclone take all of this into account in sync/copy/mkdir operations?

**Comment by 2600box** [comment]
I have this error and I have tried the suggested --exclude and --low-level-retries but the error still shows up. 

is the error preventing further progress?

**Comment by wwwindisch** [comment]
Just to share my experience (maybe it's helpful for someone...):
If I get the filter rules right, there are 2 kinds of issues:

1. Single folder or file names are too long. --> They can be filtered according to ncw's suggestion.

2. The whole path, consisting of many relatively short folder names + file name, is too long. --> Then just "????..." did not work for me, since the "?" replaces any letter except for the "/". Therefore I changed the filter to:
`/?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**?**`

This rule (length determined via trial & error) works for me for:
- OneDrive
- file name encryption
- a root folder name with 5 letters --> I assume if you sync directly to the OneDrive root you may allow a few "?**" more...

Consequently, it might make sense to have two flags:
`--max-name-length`
`--max-path-length`

**Comment by xerrni** [comment]
another corner case: If you setup second remote to e.g. other_remote:/my_folder_on_other_remote/
then standard implementation may fail.

**Comment by ivandeex** [comment]
Related to https://github.com/rclone/rclone/issues/4805

**Comment by albertony** [comment]
With the new [convmv](https://rclone.org/commands/rclone_convmv/) functionality introduced in v1.70, you now have the alternative of using option `--name-transform "all,truncate=160"` with your copy/sync/move command. This only limits the length of individual names (path elements), and you may still hit a limit for the total path length. Also, you risk that two files from the source are transformed into the same name at the destination, which may or may not be what you want.

Please correct me if I'm wrong, @nielash. :)

**Comment by nielash** [comment]
> Please correct me if I'm wrong, [@nielash](https://github.com/nielash). :)

That is all correct! I would just add that depending on your use case, `--name-transform all,truncate_bytes_keep_extension=N` might be a better fit. See https://github.com/rclone/rclone/commit/fe62a2bb4ea8f62a1567c7883c0bba81827656eb

I also think that [`--max-name-length`](https://github.com/rclone/rclone/issues/762#issuecomment-258966373) is still needed, for situations where skipping is preferable to truncating.

(Btw, writing this comment caused me to realize that the `convmv` docs are missing some options! I've submitted PR #8786 to address that.)

**Comment by ncw** [comment]
I note with the advent of regexp filters we can exclude files longer than 100 quite easily with

    --exclude "{{[^/]{101,}}}"

This could probably be extended to directories too (exercise for student ;-)
